### PR TITLE
feat: [CDS-71493]: integrate static yaml schema API's behind FF

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

`integrate static yaml schema API's behind FF`

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/72-templates-library/components/TemplateStudio/TemplateYamlView/TemplateYamlView.tsx [hunk 1]
```diff
@@ -113,9 +115,22 @@ const TemplateYamlView: React.FC = () => {
       accountIdentifier: accountId,
       scope: getScopeFromDTO({ accountIdentifier: accountId, orgIdentifier, projectIdentifier })
     },
-    lazy: isTemplateSchemaValidationDisabled
+    lazy: isTemplateSchemaValidationDisabled || STATIC_YAML_SCHEMA
+  })
+
+  const { data: templateStaticSchema } = useGetStaticSchemaYaml({
+    queryParams: {
+      entityType: 'Template',
```
Comment (PratyushGar) [inline src/modules/72-templates-library/components/TemplateStudio/TemplateYamlView/TemplateYamlView.tsx:123]: Is this entityType fixed for this schema? We have an enum for Template that can be used here and queryParams can be extracted as commonProps to be used in both

File: src/modules/70-pipeline/components/PipelineStudio/PipelineSchema/PipelineSchemaContext.tsx [hunk 2]
```diff
@@ -37,15 +40,34 @@ export function PipelineSchemaContextProvider(props: React.PropsWithChildren<unk
...
   })
+
+  const { data: pipelineStaticSchema, error: staticSchemaError } = useGetStaticSchemaYaml({
+    queryParams: {
+      entityType: 'Pipelines',
+      projectIdentifier: projectIdentifier,
+      orgIdentifier: orgIdentifier,
+      accountIdentifier: accountId,
+      scope: getScopeFromDTO({ accountIdentifier: accountId, orgIdentifier, projectIdentifier })
+    },
```
Comment (PratyushGar) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineSchema/PipelineSchemaContext.tsx:63]: Can be extracted as commonQueryParams to be used in both the placed

File: src/modules/72-templates-library/components/TemplateStudio/TemplateYamlView/TemplateYamlView.tsx (hunk 1 above)
Comment (PratyushGar) [inline src/modules/72-templates-library/components/TemplateStudio/TemplateYamlView/TemplateYamlView.tsx:123]: `entityType: template.type`

Is this an intentional change to keep entityType fixed for this schema? 
We have an enum for Template that can be used here and queryParams can be extracted as common props to be used in both places

Comment (vikrant-harness) [reply src/modules/72-templates-library/components/TemplateStudio/TemplateYamlView/TemplateYamlView.tsx:123]: Yes we need only fixed entity type now

**Review by rameshx — APPROVED** [review]

**Review by PratyushGar — DISMISSED** [review]

**Review by bhavya-sinha — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

**Review by sonaligoyal-harness — APPROVED** [review]

**Review by YounixIO — APPROVED** [review]

**Pull request merged**